machine/v1beta1/types_awsprovider.go

```diff
@@ -95,6 +95,14 @@ type AWSMachineProviderConfig struct {
 	// The field size should be greater than 0 and the field input must start with cr-***
 	// +optional
 	CapacityReservationID string `json:"capacityReservationId"`
```

Comment:
@JoelSpeed I'm not familiar with the nuances of `v1beta1` stability guarantees, but this field looks like a good candidate for some validation updates. It might be better to do ratcheting validation while this is `v1beta1` than when it is `v1`.

Reply:
Unfortunately, this API (and other ProviderSpecs as well) are used a `runtime.RawExtension`, which means there is no generated CRD schema, and, all validations we apply here, are actually implemented by a validation webhook instead.

I tend to get markers in here to represent what would be happening in the webhooks though, it's seems there are at least parts of the validations validated via a webhook, see https://github.com/openshift/machine-api-operator/blob/b397e885dafbf69f641757fea2c946fb693ae949/pkg/webhooks/machine_webhook.go#L2379-L2389

Reply:
Ah I see, good to know :). Thanks for the link